File: springboot-app/src/main/java/de/ngxa/restaurant/dto/FileAssetDto.java
package de.ngxa.restaurant.dto;

import de.ngxa.restaurant.entity.AlbumCategory;
import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
public class FileAssetDto extends BaseDto {

	@NotNull
	private AlbumCategory category;

	@NotEmpty
	private String fileName;
	private long fileSize;
	private String fileUrl;
	private String rawName;
	private String extension;
	private String fileType;
	@NotEmpty
	private byte[] content;

	private String hashFileName;
}

File: manager-app/src/app/dto/file-asset-dto.ts
import {BaseDto} from './base-dto';
import {Album} from './album';

export class FileAssetDto extends BaseDto {

  category: Album;
  fileName:  string;
  fileSize: number;
  fileUrl: string;
  rawName: string;
  extension: string;
  fileType: string;
  content: string;

  hashFileName: string;
}
